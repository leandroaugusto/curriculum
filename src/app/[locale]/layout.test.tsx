import { render, screen } from "@testing-library/react";

import RootLayout from "./layout";

jest.mock("next/navigation", () => ({
  notFound: () => <div data-testid="mock-not-found" />,
}));

jest.mock("@/constants/locales", () => ({
  locales: ["en", "pt"],
}));

describe("RootLayout", () => {
  it("renders the RootLayout", () => {
    const { container } = render(
      <RootLayout params={{ locale: "en" }}>
        <p>Test</p>
      </RootLayout>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render notFound page when locale wasn't found", () => {
    render(
      <RootLayout params={{ locale: "es" }}>
        <p>Test</p>
      </RootLayout>
    );

    const notFoundPage = screen.getByTestId("mock-not-found");

    expect(notFoundPage).toBeInTheDocument();
  });

  it("should render HTML's lang equal pt-BR", () => {
    render(
      <RootLayout params={{ locale: "pt" }}>
        <p>Test</p>
      </RootLayout>
    );
    const root = screen.getByTestId("root-layout");

    expect(root).toHaveAttribute("lang", "pt-BR");
  });
});
