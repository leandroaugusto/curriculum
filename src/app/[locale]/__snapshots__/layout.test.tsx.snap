// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout renders the RootLayout 1`] = `
<div>
  <html
    class="h-full"
    data-testid="root-layout"
    lang="en-GB"
  >
    <body
      class="className bg-white min-h-full flex flex-col"
    >
      <p>
        Test
      </p>
    </body>
  </html>
</div>
`;
